src/sentry/search/events/builder.py

```diff
@@ -1552,6 +1559,138 @@ def get_snql_query(self) -> None:
+                    for column in self.columns
+                    if not isinstance(column, CurriedFunction) or column in query_details.functions
+                ]
+                if groupby_values:
+                    # We already got the groupby values we want, add them to the conditions to limit our results so we
+                    # can get the aggregates for the same values
+                    where = self.where + [
+                        Condition(
+                            # Tuples are allowed to have multiple types in clickhouse
+                            Function(
+                                "tuple",
+                                [
+                                    groupby.exp
+                                    if isinstance(groupby, AliasedExpression)
+                                    else groupby
+                                    for groupby in self.groupby
+                                ],
+                            ),
+                            Op.IN,
+                            Function("tuple", groupby_values),
+                        )
+                    ]
+                    # Because we've added a condition for each groupby value we don't want an offset here
+                    offset = Offset(0)
+                else:
+                    # We don't have our groupby values yet, this means this is the query where we're getting them
+                    where = self.where
+                    offset = self.offset
+
+                query = Query(
+                    dataset=self.dataset.value,
+                    match=query_details.entity,
+                    select=select,
+                    array_join=self.array_join,
+                    where=where,
+                    having=self.having,
+                    groupby=self.groupby,
+                    orderby=query_details.orderby,
+                    limit=self.limit,
+                    offset=offset,
+                    limitby=self.limitby,
+                    turbo=self.turbo,
+                )
+                current_result = raw_snql_query(
+                    query,
+                    referrer,
+                    use_cache,
+                )
+                for row in current_result["data"]:
+                    # Arrays in clickhouse cannot contain multiple types, and since groupby values
+                    # can contain any type, we must use tuples instead
+                    groupby_key = tuple(row[key] for key in groupby_aliases)
+                    value_map_key = ",".join(str(value) for value in groupby_key)
+                    # First time we're seeing this value, add it to the values we're going to filter by
+                    if value_map_key not in value_map:
+                        groupby_values.append(groupby_key)
+                    value_map[value_map_key].update(row)
+                result["meta"] += current_result["meta"]
+        result["data"] = list(value_map.values())
```

Comment:
Something I thought of: metrics makes no guarantees that there will be data for every aggregation for every group by. This is a contrived example, but there could be a p95 for transaction `x` but not unique users. Is your code supposed to handle that case?

Reply:
yes :+1: I'll add an explicit test for this, but what should happen is that we'll get p95 for `x`, and nothing for count_unique.

Reply:
Ah, very good point, i just realized for this scenario if we sort by unique users we'll get an unexpected result (`x` just won't show up at all), and I don't think there's a workaround either 🤔 Going to document this as a known deficiency for now.